### DISCUSSION TITLE
post route not found

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [rwf2/Rocket#2426] [Questions] [answered]
I am not sure if I am doing something wrong. I have try to get a post route working. 
But testing with curl have failed every time with a 404.
I am also confused why the route should be defined twice.
One time at `#[post("/register", data = "<token>")]` and once at `.mount("/register", routes![register])`.
I can even set different values here.

```bash
❯ curl --data 1234 "http://127.0.0.1:8000/register"
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>404 Not Found</title>
</head>
<body align="center">
    <div role="main" align="center">
        <h1>404: Not Found</h1>
        <p>The requested resource could not be found.</p>
        <hr />
    </div>
    <div role="contentinfo" align="center">
        <small>Rocket</small>
    </div>
</body>
</html>
```
rocket log:
```
POST /register application/x-www-form-urlencoded:
   >> No matching routes for POST /register application/x-www-form-urlencoded.
   >> No 404 catcher registered. Using Rocket default.
   >> Response succeeded.
```
my code:
`user.rs`
```rust
use rocket;
use rocket::http::Status;
use crate::CONFIG;

#[post("/register", data = "<token>")]
pub(crate) async fn register(token: String) -> Result<String, Status> {
    if token==CONFIG.register_token {
        return Ok("TODO".to_owned())
    }
    Err(Status::Unauthorized)
}
```

`main.rs`
```rust
#![warn(rust_2018_idioms)]
#![forbid(unsafe_code)]

#[macro_use]
extern crate rocket;
use rocket::{http::Status, shield::Shield, tokio::task::spawn_blocking};
use rocket_dyn_templates::{context, Template};
...
mod user;
use user::*;

...

#[launch]
async fn rocket() -> _ {
	...
	rocket::build()
		.mount("/", routes![index])
		.mount("/register", routes![register])
		.attach(Template::fairing())
		.attach(shield)
}
```
Full code: https://github.com/LuckyTurtleDev/mstickerpicker/tree/7bb874f35a7036026d8bd4cc27eb88d6af325748

**Comment by propellaone** [comment, marked as answer]
Using this line 
`.mount("/register", routes![register])`
the route will be located at "/register/register". I guess you're expecting it to be found at "/register".
So, using
`.mount("/", routes![index, register])`
should probably work.

**Reply by LuckyTurtleDev** [reply]
Thanks this was the issue.
So the route will  be prefixed with the value from `.mount()`. I  have already wondered, why their was 2 route paths.
